demisto_sdk/runners/playbook_runner.py

```diff
@@ -0,0 +1,123 @@
+
+            while elasped_time < self.timeout:
+                playbook_results = self.get_playbook_results_dict(incident_id)
+                if playbook_results["state"] == "inprogress":
+                    time.sleep(10)
+                    elasped_time = int(time.time() - start_time)
+                else:   # the playbook has finished running
+                    break
+
+            # Ended the loop because of timeout
+            if elasped_time >= self.timeout:
+                print_error(f'The command had timed out while the playbook is in progress.\n'
+                            f'To keep tracking the playbook please go to : {work_plan_link}')
+            else:
+                if playbook_results["state"] == "failed":
+                    print_error("The playbook finished running with status: FAILED")
+                else:
+                    print_color("The playbook has completed it's run successfully", LOG_COLORS.GREEN)
+
+        # The command does not wait for the playbook to finish running
+        else:
+            print(f'To see results please go to : {work_plan_link}')
+
+        return 0
+
+    def create_incident_with_playbook(self, incident_name, playbook_id):
+        # type: (str, str) -> int
+        """Create an incident in Demisto with the given incident_name and the given playbook_id
+
+        Args:
+            incident_name (str): The name of the incident
+            playbook_id (str): The id of the playbook
+
+        Returns:
+            int. The new incident's ID. Returns 1 in a case of creation error.
+        """
+
+        create_incident_request = demisto_client.demisto_api.CreateIncidentRequest()
+        create_incident_request.create_investigation = True
+        create_incident_request.playbook_id = playbook_id
+        create_incident_request.name = incident_name
+
+        try:
+            response = self.demisto_client.create_incident(create_incident_request=create_incident_request)
+        except ApiException:
+            print_error(f'Failed to create incident with playbook id : "{playbook_id}", '
+                        'possible reasons are:\n'
+                        '1. This playbook name does not exist \n'
+                        '2. Schema problems in the playbook \n'
+                        '3. Unauthorized api key')
+            return -1
+
+        print_color(f'The playbook: {self.playbook_id} was triggered successfully.', LOG_COLORS.GREEN)
+        return response.id
+
+    def get_playbook_results_dict(self, inc_id):
+        playbook_results = self.demisto_client.generic_request(method='GET', path=f'/inv-playbook/{inc_id}')
+        return eval(playbook_results[0])
+
+    def get_api_key(self):
+        """Retrieve the API Key
+
+        Raises:
+            RuntimeError: if the API Key environment variable is not found
+
+        Returns:
+            str: API Key
+        """
+        api_key = os.environ.get(self.DEMISTO_API_KEY_ENV, None)
```

Comment:
```suggestion
        api_key = os.environ.get(self.DEMISTO_API_KEY_ENV)
```

Reply:
Also, are we sure the API key will be there?
What if we'll start using this logic for running nightly, or a local content build?

Reply:
then we will have to configure this env variable